### PR TITLE
huuuuge commit for huuuuge pull request

### PR DESCRIPTION
**Pull request by danielm1405**

### CODE REVIEW

File: build_and_run_tests.bash [hunk 1]
```diff
@@ -6,3 +6,5 @@ cd test
 cmake CMakeLists.txt
 make
 ./runTests
+
+rm -r CMakeCache.txt CMakeFiles/ Makefile cmake_install.cmake runTests
```
Comment (jul3x) [inline build_and_run_tests.bash:10]: why?

File: include/EvolutionaryAlgorithm/CarParameters.h [hunk 2]
```diff
@@ -12,13 +15,51 @@
...
+    //!  A CarParameters default constructor.
+    CarParameters() = default;
 
+    //! A CarParameters constructor.
+    /*!
+      Constructor that specifies front and rear wheel radius
+      shape of the car body and position of rotation axis of wheels
+    */
+    CarParameters(double front_wheel_radius, double rear_wheel_radius,
+                  const std::vector<b2Vec2>& car_body, b2Vec2 front_joint, b2Vec2 rear_joint);
```
Comment (jul3x) [inline include/EvolutionaryAlgorithm/CarParameters.h:27]: const reference on joints?

File: include/EvolutionaryAlgorithm/CarParameters.h [hunk 3]
```diff
@@ -12,13 +15,51 @@
 class CarParameters {
```
Comment (jul3x) [inline include/EvolutionaryAlgorithm/CarParameters.h:15]: this is structure actually

File: include/EvolutionaryAlgorithm/EvolutionaryAlgorithm.h [hunk 4]
```diff
@@ -1,34 +1,169 @@
...
+    const std::vector<CarParameters> generateNewPopulation();
+    //!  A Method for getting the value of mutation probability in newly generated populations.
+    const double getMutationProbability() const;
+    //!  A Method for getting the value of crossover probability in newly generated populations.
     const double getCrossoverProbability() const;
-    void setMutationProbablity_(double mutation_probablity);
+    //!  A Method for setting the value of mutation probability in newly generated populations.
+    void setMutationProbability_(double mutation_probability);
+    //!  A Method for setting the value of crossover probability in newly generated populations.
     void setCrossoverProbability_(double crossover_probability);
```
Comment (jul3x) [inline include/EvolutionaryAlgorithm/EvolutionaryAlgorithm.h:49]: _?

**Review by jul3x — APPROVED** [review]

File: build_and_run_tests.bash (hunk 1 above)
Comment (danielm1405) [reply build_and_run_tests.bash:10]: it left mess

File: include/EvolutionaryAlgorithm/CarParameters.h (hunk 2 above)
Comment (danielm1405) [reply include/EvolutionaryAlgorithm/CarParameters.h:27]: :scream:

File: include/EvolutionaryAlgorithm/CarParameters.h (hunk 3 above)
Comment (danielm1405) [reply include/EvolutionaryAlgorithm/CarParameters.h:15]: right

File: include/EvolutionaryAlgorithm/EvolutionaryAlgorithm.h (hunk 4 above)
Comment (danielm1405) [reply include/EvolutionaryAlgorithm/EvolutionaryAlgorithm.h:49]: dunno

**Review by danielm1405 — COMMENTED** [review]
@jul3x merge??

**Comment by jul3x** [comment]
> 
> 
> @jul3x merge??

yes

**Pull request merged**